CONTRIBUTING.md

```diff
@@ -3,11 +3,70 @@ Guidelines for Contributing
 
 Welcome to the D community and thanks for your interest in contributing! To get started, please read the [Starting as a Contributor](http://wiki.dlang.org/Starting_as_a_Contributor) article on the D Wiki.
 
-More Links
-----------
-
-* Fork [on Github](https://github.com/D-Programming-Language/phobos)
-* Use our [Bugzilla bug tracker](http://d.puremagic.com/issues/)
-* Follow the [Styleguide](http://dlang.org/dstyle.html)
-* Participate in [our forum](http://forum.dlang.org/)
-* Review Phobos additions in the [Review Queue](http://wiki.dlang.org/Review_Queue).
\ No newline at end of file
+Quick links
+-----------
+
+- Fork [on Github](https://github.com/D-Programming-Language/phobos)
+- Use our [Bugzilla bug tracker](http://d.puremagic.com/issues/)
+- Follow the [Styleguide](http://dlang.org/dstyle.html)
+- Participate in [our forum](http://forum.dlang.org/)
+- Ask questions on `#d` ([Web interface](https://kiwiirc.com/client/irc.freenode.net/d))
+- Review Phobos additions in the [Review Queue](http://wiki.dlang.org/Review_Queue)
+
+How to make a good submission?
+------------------------------
+
+The usual workflow is:
+
+1. [Fork](https://help.github.com/articles/fork-a-repo/) [Phobos](https://github.com/D-Programming-Language/phobos) on Github
+2. Create your [own branch](https://github.com/Kunena/Kunena-Forum/wiki/Create-a-new-branch-with-git-and-manage-branches)
+3. Work locally on your new feature or fix (see the tips below)
+4. Submit your [pull request](https://help.github.com/articles/creating-a-pull-request/)
+
+Below is a list of tips and checks that you should consider before opening your PR:
+
+### Code
+
+- Don't add `@nogc`, `@pure`, `@safe`, `nothrow` attributes yourself - let the compiler infer it! However add unittest check for these attributes if possible
+- Use nested structs for ranges (aka [Voldemort types](http://wiki.dlang.org/Voldemort_types))
+- Avoid to use `auto` as return type wherever possible (makes the documentation easier to read)
+- Don't be afraid to use the new features (e.g. `Foo!Range` vs. `Foo!(Range)`) - some parts of Phobos are a bit older and have some relicts due to lacking functionality of the compiler
+- Be critical when looking at Phobos code
+- Don't mix bug fixes with enhancements or refactoring
+- Bigger changes (rule of thumb: >500 LoC) have to go to `std.experimental` first.
```

Comment:
Only big additions like new modules should to `std.experimental` after they pass the [review process](http://wiki.dlang.org/Review/Process). Smaller additions like individual function can go directly, after @andralex approves them.
Big deletions and refactoring just need OK from two other reviewers.